common/src/main/java/org/jboss/hal/testsuite/util/ConfigChecker.java

```diff
@@ -0,0 +1,145 @@
+     */
+    public ResourceVerifier verifyFormNotSaved() throws Exception {
+        Assert.assertFalse("Configuration should NOT switch into read-only mode.", saved);
+        return new ResourceVerifier(resourceAddress, client);
+    }
+
+    private ConfigChecker(Builder builder)
+            throws IOException, InterruptedException, TimeoutException {
+        this.client = builder.client;
+        this.resourceAddress = builder.resourceAddress;
+        Object attrValue = getAttrValue(builder.attrValue, builder.inputType);
+        edit(builder.config, builder.inputType, builder.identifier, attrValue);
+    }
+
+    private Object getAttrValue(Object attrValue, InputType inputType) {
+        switch (inputType) {
+            case TEXT: case SELECT:
+                if (attrValue instanceof String) {
+                    return attrValue;
+                } else if (attrValue instanceof Long) {
+                    return String.valueOf((long) attrValue);
+                } else if (attrValue instanceof Integer) {
+                    return String.valueOf((int) attrValue);
+                } else {
+                    throw new IllegalArgumentException(attrValue + " should be String, Integer or Long!");
+                }
+            case CHECKBOX:
+                if (attrValue instanceof Boolean) {
+                    return attrValue;
+                } else {
+                    throw new IllegalArgumentException(attrValue + " should be Boolean!");
+                }
+            default:
+                throw new IllegalArgumentException("Not yet supported inputType: " + inputType);
+        }
+    }
+
+    private void edit(ConfigFragment config, InputType inputType, String identifier, Object attrValue) throws IOException, InterruptedException, TimeoutException {
+        switch (inputType) {
+            case TEXT:
+                saved = config.editTextAndSave(identifier, (String) attrValue); break;
+            case CHECKBOX:
+                saved = config.editCheckboxAndSave(identifier, (Boolean) attrValue); break;
+            case SELECT:
+                saved = config.selectOptionAndSave(identifier, (String) attrValue); break;
+        }
+        if (!saved) {
+            config.cancel(); // cleanup
+        }
+    }
+
+    public enum InputType {
+        TEXT, SELECT, CHECKBOX;
+    }
+
+    public static class Builder {
+        private final OnlineManagementClient client;
+        private final Address resourceAddress;
+        private ConfigFragment config;
+        private InputType inputType;
+        private String identifier;
+        private Object attrValue;
+
+        public Builder(OnlineManagementClient client, Address resourceAddress) {
+            this.client = client;
+            this.resourceAddress = resourceAddress;
+        }
+
+        /**
+         * setter for {@link ConfigFragment}
+         */
+        public Builder config(ConfigFragment config) {
```

Comment:
Maybe, I would change the name to configFragment to make it more clear that the UI element is expected.